### PR TITLE
[DOCFIX] Fix colon in yml files

### PR DESCRIPTION
**Pull request by bf8086**
We recently added colon marks in yml file which broke the syntax. Put them in quotes fixed it.

### CODE REVIEW

**Comment by apc999** [comment]
@bf8086 instead of fixing `docs/_data/table/en/worker-configuration.yml`, can you actually fix the text in `PropertyKey.java` which is the source of content in `worker-configuration.yml`

**Comment by bf8086** [comment]
@apc999 The problem is due to the limitation of raw string representation in YAML format. I updated `ConfigurationDocGenerator.java` to quote the description strings and escape them correctly.

**Comment by apc999** [comment]
can you fix the tests? thanks

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java [hunk 1]
```diff
@@ -153,10 +153,10 @@ static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, String f
 
       for (PropertyKey iteratorPK : dfkeys) {
         String pKey = iteratorPK.toString();
-        String description = iteratorPK.getDescription();
+        String description = iteratorPK.getDescription().replace("'", "''");
 
         // Write property key and default value to yml files
-        String keyValueStr = pKey + ":\n  " + description + "\n";
+        String keyValueStr = pKey + ":\n  '" + description + "'\n";
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: NIT: add comment why it has to be quoted.

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java [hunk 2]
```diff
@@ -153,10 +153,10 @@ static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, String f
 
       for (PropertyKey iteratorPK : dfkeys) {
         String pKey = iteratorPK.toString();
-        String description = iteratorPK.getDescription();
+        String description = iteratorPK.getDescription().replace("'", "''");
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: This is the correct way to escape `'` for jekyll?

Comment (bf8086) [inline shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: This is to escape `'` for single quoted strings in YAML files. YAML files supports multiple ways to declare a string literal, and single quoted string has the least requirement in format or escaped characters. The original issue was that we write description in unquoted strings where jekyll complained about `:` in them.

You can check out the page [here](http://symfony.com/doc/current/components/yaml/yaml_format.html#strings) for a comparison of different string representations.

File: shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java (hunk 1 above)
Comment (bf8086) [inline shell/src/main/java/alluxio/cli/ConfigurationDocGenerator.java]: Done.

**Pull request merged**